waltid-services/waltid-verifier-api/src/main/kotlin/id/walt/verifier/VerifierApi.kt

```diff
@@ -454,43 +455,41 @@ private fun getErrorDescription(it: Throwable): String? = when (it.localizedMess
     else -> null
 }
 
-private suspend fun PipelineContext<Unit, ApplicationCall>.processError(
-    sessionId: String, exception: Throwable
-) {
-    val session = verificationUseCase.getSession(sessionId)
-    if (session.walletInitiatedAuthState != null) {
-        val state = session.walletInitiatedAuthState
-        context.respondRedirect(
-            "openid://?state=$state&error=invalid_request&error_description=${getErrorDescription(exception)}"
-        )
-    }
-}
-
 private suspend fun PipelineContext<Unit, ApplicationCall>.processVerificationFailureResult(
-    it: Throwable,
-    sessionId: String?
+    sessionId: String?,
+    error: Throwable,
 ) {
-    logger.debug(it) { "Verification failed ($it)" }
-    val errorDescription = it.localizedMessage
+    logger.debug(error) { "Verification failed ($error)" }
+    val errorDescription = error.message ?: "Verification failed"
     logger.error { "Error: $errorDescription" }
     if (sessionId != null) {
-        processError(sessionId, it)
+        val session = verificationUseCase.getSession(sessionId)
```

Comment:
this could be extracted into a separate method, e.g. processError:
```kotlin
private suspend fun PipelineContext<Unit, ApplicationCall>.processError(
    sessionId: String,
    error: Throwable
) {
    val session = verificationUseCase.getSession(sessionId)
    if (session.walletInitiatedAuthState != null) {
        val state = session.walletInitiatedAuthState
        context.respondRedirect(
            "openid://?state=$state&error=invalid_request&error_description=${getErrorDescription(error)}"
        )
    } else if (error is VerificationUseCase.FailedVerificationException && error.redirectUrl != null) {
        context.respond(HttpStatusCode.BadRequest, error.redirectUrl)
    } else {
        throw error
    }
}
```
then `processVerificationFailureResult` would look like:
```kotlin
private suspend fun PipelineContext<Unit, ApplicationCall>.processVerificationFailureResult(
    sessionId: String?,
    error: Throwable,
) {
    logger.debug(error) { "Verification failed ($error)" }
    val errorDescription = error.message ?: "Verification failed"
    logger.error { "Error: $errorDescription" }
    if (sessionId != null) {
        processError(sessionId, error)
    } else {
        logger.error(error) { "/verify error: $errorDescription" }
        call.respond(HttpStatusCode.BadRequest, errorDescription)
    }
}
```